src/main/resources/de/eddyson/tapestry/react/services/isomorphic/r.js-README.md

```diff
@@ -0,0 +1,47 @@
+# r.js incompatibilities
+
+Date: 2017-01-05
+
+As of writing this, the r.js library in it's current version (2.3.2) has an incompatibility with nashorn (Oracle Java 8 1.8.0_111) interpreter:
+
+This problematic code is starting in line 300 of r.js:
+
+```javascript
+/**
+ * Helper function for iterating over an array. If the func returns
+ * a true value, it will break out of the loop.
+ */
+function each(ary, func) {
+    if (ary) {
+        var i;
+        for (i = 0; i < ary.length; i += 1) {
+            if (ary[i] && func(ary[i], i, ary)) {
+                break;
+            }
+        }
+    }
+}
+```
+
+This code will throw an AssertionError with the message 'duplicate code' when running with -ae (assertions enabled) as
+soon as the require() function is invoked.
+
+We needed to patch the function:
+
+```javascript
+/**
+ * Helper function for iterating over an array. If the func returns
+ * a true value, it will break out of the loop.
+ */
+function each(ary, func) {
+    if (ary) {
+        var i;
+        for (i = 0; i < ary.length; i += 1) {
+            if (ary[i] && func.apply(null, [ary[i], i, ary])) {
+                break;
+            }
+        }
+    }
+}
+```
+The change is only syntactic, from `func(ary[i], i, ary)` to `func.apply(null, [ary[i], i, ary])`
```

Comment:
I don't think that this is a syntactic change only. AFAICT, `func(ary[i], i, ary)` is equivalent to `func.call(this, ary[i], i, ary)` or `func.apply(this, [ary[i], i, ary])`.
But do you have any clue as to what happens exactly? Why does it work with the change?
I must admit that I don't like shipping a patched version of `r.js`. Usually, those things are hard to maintain in the future. But since you documented it so well, we can probably make an exception here. However, sooner or later, I'd really like to have this fixed than worked around.